foo_uie_console/main.cpp

```diff
@@ -267,6 +282,9 @@ void ConsoleWindow::set_config(stream_reader* reader, t_size p_size, abort_callb
         int32_t edge_style{};
         reader->read_lendian_t(edge_style, abort);
         m_edge_style = EdgeStyle{edge_style};
+        int32_t append_newline{};
+        reader->read_lendian_t(append_newline, abort);
```

Comment:
It would be good to handle this more gracefully on upgrade from an older version of the panel where this value won't be in the stream (right now there's an error being logged to the console).

I would typically do something like this:

```cpp
        try {
            reader->read_lendian_t(m_append_newline, abort);
        } catch (const exception_io_data_truncation&) {
            m_append_newline = true;
        }
```

Reply:
Oh, completely missed that.

Reply:
...however, is setting `m_append_newline` in catch clause absolutely necessary? Won't its initial value suffice or am I misunderstanding usage of these `cfg_var`s?

Reply:
Ah, I did mean `m_append_newline = cfg_last_hide_trailing_newline`, but, yes, it should be able to go since `m_append_newline ` is initialised correctly anyway.